packages/zoe/tools/types.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @typedef {Object} TimerService Gives the ability to get the current time,
+ * schedule a single wake() call, create a repeater that will allow scheduling
+ * of events at regular intervals, or remove scheduled calls.
+ * @property {() => Timestamp} getCurrentTimestamp Retrieve the latest timestamp
+ * @property {(baseTime: Timestamp, handler: TimerServiceHandler) => Timestamp}
+ * setWakeup Return value is the time at which the call is scheduled to take
+ * place
+ * @property {(handler: TimerServiceHandler) => Array<Timestamp>} removeWakeup Remove
+ * the handler from all its scheduled wakeup, whether produced by
+ * `timer.setWakeup(h)` or `repeater.schedule(h)`.
+ * @property {(baseTime: Timestamp, interval: RelativeTime) => TimerRepeater}
+ * createRepeater Create and return a repeater that will schedule `wake()` calls
+ * repeatedly at times that are a multiple of interval following baseTime.
+ * Interval is the delay between successive times at which wake will be called.
+ * When `schedule(h)` is called, `h.wake()` will be scheduled to be called after
+ * the next multiple of interval from the base. Since times can be
+ * coarse-grained, the actual call may occur later, but this won't change when
+ * the next event will be called.
+ */
+
+/**
+ * @typedef {number} Timestamp An absolute individual stamp returned by a
+ * TimerService.  Note that different timer services may have different
+ * interpretations of actual Timestamp values.
+ * @typedef {number} RelativeTime Difference between two Timestamps.  Note that
+ * different timer services may have different interpretations of actual
+ * RelativeTime values.
+ */
+
+/**
+ * @typedef {Object} TimerServiceHandler
+ * @property {(timestamp: Timestamp) => void} wake The timestamp passed to
+ * `wake()` is the time that the call was scheduled to occur.
+ */
+
+/**
+ * @typedef {Object} TimerRepeater
+ * @property {(handler: TimerServiceHandler) => void} schedule Returns the time
+ * scheduled for the first call on handler.  The handler will continue to be
+ * scheduled for a `wake()` call every interval until the repeater is disabled.
+ * @property {() => void} disable Disable this repeater, so `schedule()` can't
+ * be called, and handlers already scheduled with this repeater won't be
+ * rescheduled again after `wake()` is next called on them.
+ */
+
+/**
+ * @typedef {Object} PriceQuote
+ * @property {Amount} quoteAmount Amount whose value is a PriceQuoteValue
+ * @property {Payment} quotePayment The `quoteAmount` wrapped as a payment
+ */
+
+/**
+ * @typedef {Array<PriceDescription>} PriceQuoteValue A set of PriceDescriptions
```

Comment:
Why is it a set? If it's because the PriceQuote's amount uses SetMath, then I think those amounts should contain sets of single PriceDescriptions. Otherwise, I'm confused by this.

Reply:
I went back and forth on this several times, but I agree with you.  I've made it a "single-valued set" with the appropriate type.